### ISSUE TITLE
How to handle multiple .require() in a session

### ISSUE THREAD

**Issue by peol** [addyosmani/basket.js#29]
Currently, we don't handle this at all, we always execute the script for every `.require()`, which means that if we do:

``` javascript
basket.require({ url: 'jquery.js' }).wait(function() {
  basket.require({ url: 'jquery.js' });
});
```

This will make us insert a new `<script>` for each, overwriting any plugins loaded in-between, internal jquery caches etc. Should we just let this be, or should we track this internally and not insert the script more than once?

**Comment by addyosmani** [comment]
I think we should handle this internally, but I'm wondering if it would make sense to also make this behaviour configurable (i.e have a sane default but one that can be overridden).

**Comment by sindresorhus** [comment]
Yes, I also think we should handle this, but it's probably smart to have a way to override/disable.

**Comment by peol** [comment]
Great. Would be great if anyone would have time to fix this, I'm working on the TTL stuff as we speak and the tests doesn't pass because of this. :D

**Comment by addyosmani** [comment]
Might not be able to take a look at it today but can definitely do it this week! As long as no one beats me to it :p

**Comment by sindresorhus** [comment]
"definitely do it this week" - you said it :p

When would anyone want to load a script twice? Just trying to come up with a reason it should be configurable. And why would anyone try to load the same script twice in the first place?

**Comment by peol** [comment]
I think the problem is the other way around. Currently, if the dev by mistake `basket.require()` the same script more than once, we'll write over the old one (in this example, let's say he pulled in jQuery and then added some plugins, and then `basket.require()` jQuery again), he'll lose all the plugins. I'm not sure that's something we want to handle, but it might be nice from both this perspective, and possibly even for debugging (getting access to a list of what files that has been loaded, which we use to see if a script has been loaded already).

**Comment by sindresorhus** [comment]
I think we should prevent it by default. If someone wants (for whatever insane reason) to load a script twice they could just `basket.remove()` it first right?

**Comment by Startouf** [comment]
So no news on that ? My website is using Rails and Turbolinks (loading pages with AJAX) and I would have expected require() to be ignored if the current page was reloaded using turbolinks with scripts already loaded once
